File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.java
import java.util.*;

class Solution {
    class Pair{
        int a, r; boolean b;
        Pair(int a, int r, boolean b){
            this.a = a;
            this.r = r;
            this.b = b;
        }
    }
    private int stuNumber(int a, int[] rank){
        for(int i = 0; i < rank.length; i++){
            if(a == rank[i]) return i;
        }
        return 0;
    }
    public int solution(int[] rank, boolean[] attendance) {
        List<Pair> l = new ArrayList<>();
        for(int i = 0; i < rank.length; i++){
            l.add(new Pair(rank[i], i, attendance[i]));
        }
        Collections.sort(l, (a, b)-> Integer.compare(a.a, b.a));
        int[] ret = new int[3];
        int idx = 0;
        for(Pair p : l){
            if(ret[2] != 0) break;
            if(p.b) ret[idx++] = p.r;
        }
        return (10000 * ret[0]) + (100 * ret[1]) + ret[2];
    }
}

File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.py
def solution(rank, attendance):
    idxs = [x for x in range(len(rank))]
    print(idxs)
    idxs.sort(key=lambda x: rank[x])
    ret = []
    idx = 0
    print(idxs)
    while len(ret) < 3 and idx < len(idxs):
        if attendance[idxs[idx]]:
            ret.append(idxs[idx] )
        idx += 1
    a, b, c = ret
    
    return 10000 * a + 100 * b + c
    
"""
참여가능 중 3명 선발
idx를 숫자 높은 순으로 정렬해서 참석가능한 3명뽑고 거기서 계산하기
"""